### DISCUSSION TITLE
Idea for Feature  - Pass Keys Via Cli

### DISCUSSION THREAD

**Discussion by RandomRobbieBF** [BishopFox/cloudfox#38] [Ideas]
Hey,

Can we add an option for passing a key and secret via the CLI to test if they work instead of having to set up profiles as i have a list of keys / secrets i want to be able to test but would have to set up a profile for each one.

Regards,
Robbie

**Comment by sethsec-bf** [comment]
Hey @RandomRobbieBF  - Thanks for the feedback. 

If you set the KEY and secret as environment variables, cloudfox will just use them transparently exactly like the aws cli will use them. The trick is to just not specify a profile name at all with the aws cli and cloudfox: 

```
❯ export AWS_ACCESS_KEY_ID=AKIAEXAMPLE
❯ export AWS_SECRET_ACCESS_KEY=harXmptWKEXAMPLE
❯ aws sts get-caller-identity
❯ cloudfox aws inventory -v2
```

Does that help with your situation, or do you still think we should throw in the option to just pass them at the CLI also?

**Comment by RandomRobbieBF** [comment]
Hey @sethsec-bf id probably say it's best to add the argument to the cli to ensure no confusion. 

this would allow to iterate over the mountain of keys i need to test but would have to figure a way out to name it so when it outputs i can track which key it was.
